File: plugins/jd/skills/error-audit/examples/test-projects/go-broken/payment/charge.go
package payment

import (
	"context"
	"fmt"
	"log"
)

func ChargeCustomer(ctx context.Context, customerID string, amount int, password string) error {
	// VIOLATION _common/01: log password
	log.Printf("charging customer %s with password %s amount %d", customerID, password, amount)

	// VIOLATION go/01: ignored error return
	_ = chargeProvider(amount)

	result, err := saveTransaction(customerID, amount)
	if err != nil {
		// VIOLATION go/02: no %w wrapping
		return fmt.Errorf("save transaction failed: " + err.Error())
	}

	// VIOLATION go/03: goroutine without recover
	go func() {
		notifyCustomer(customerID, result)
	}()

	return nil
}

func chargeProvider(amount int) error            { return nil }
func saveTransaction(id string, amt int) (any, error) { return nil, nil }
func notifyCustomer(id string, r any)            {}

File: plugins/jd/skills/error-audit/examples/test-projects/nodejs-nestjs-broken/src/payment/charge.ts
import { logger } from '../utils/logger';

export async function chargeCustomer(customerId: string, amount: number, paymentMethodToken: string) {
  // VIOLATION nodejs/01: .then without .catch
  return chargeProvider.charge(amount, paymentMethodToken).then(result => saveTransaction(result));
}

export async function refundCustomer(transactionId: string) {
  try {
    await chargeProvider.refund(transactionId);
  } catch (e) {
    // VIOLATION _common/02: swallow catch (loses stack + context)
    console.log(e.message);
  }
}

export async function logPayment(customerId: string, password: string, amount: number) {
  // VIOLATION _common/01: log sensitive data (password)
  logger.info('payment attempt', { customerId, password, amount });
}

declare const chargeProvider: any;
declare function saveTransaction(r: any): any;
